knowledge-base/grid-state-filter-nullable-bool.md

```diff
@@ -0,0 +1,112 @@
+---
+title: Programatically filter a nullable bool column in the Grid by null values
+description: Learn how to programmatically filter nullable bool values in the Grid component for Blazor.
+type: how-to
+page_title: How to filter a Grid column bound to a nullable bool by the null values programatically
+slug: grid-kb-state-filter-nullable-bool
+tags: grid, blazor, column, datagrid, filter, state, nullable, bool
+res_type: kb
+ticketid: 1658561
+---
+
+## Environment
+
+<table>
+    <tbody>
+        <tr>
+            <td>Product</td>
+            <td>Grid for Blazor</td>
+        </tr>
+    </tbody>
+</table>
+
+## Description
+
+I want to programatically filter a Grid column that is bound to `bool?` by the `null` values. 
+
+
+This KB article also answers the following questions:
+- How to filter a Grid column bound to a `bool?` by the null values programatically
+
+## Solution
+
+To filter a Grid column bound to a `bool?` by the null values programatically:
+
+````CSHTML
+@using Telerik.DataSource;
+
+<TelerikButton ThemeColor="primary" OnClick="@SetGridFilter">set filtering from code</TelerikButton>
+
+<TelerikGrid Data="@MyData" Height="400px" @ref="@GridRef"
+             Pageable="true" FilterMode="@GridFilterMode.FilterMenu">
+    <GridColumns>
+        <GridColumn Field="@(nameof(SampleData.Id))" Width="120px" />
+        <GridColumn Field="@(nameof(SampleData.Name))" Title="Employee Name" />
+        <GridColumn Field="@(nameof(SampleData.Team))" Title="Team" />
+        <GridColumn Field="@(nameof(SampleData.IsOnLeave))" Title="On Leave" />
+        <GridColumn Field="@(nameof(SampleData.HireDate))" Title="Hire Date" />
+    </GridColumns>
+</TelerikGrid>
+
+@code {
+    private TelerikGrid<SampleData> GridRef { get; set; }
```

Comment:
By default, new Blazor apps use `<Nullable>enable</Nullable>` and this line will generate a warning. The best practice is to make the variable nullable and check for nullability wherever the code uses `GridRef`. See https://docs.telerik.com/blazor-ui/components/grid/filter/searchbox#search-from-code

```suggestion
    private TelerikGrid<SampleData>? GridRef { get; set; }
```